lms/djangoapps/instructor/enrollment.py

```diff
@@ -267,6 +272,29 @@ def reset_student_attempts(course_id, student, module_state_key, requesting_user
 
     if delete_module:
         module_to_reset.delete()
+        if block and block.has_score:
+            cache = FieldDataCache.cache_for_descriptor_descendents(
+                course_id=course_id,
+                user=student,
+                descriptor=block,
+                depth=0
+            )
+            module = get_module_for_descriptor(
+                user=student,
+                request=get_request(),
+                descriptor=block,
+                field_data_cache=cache,
+                course_key=course_id
+            )
+            _, points_possible = weighted_score(0, module.max_score(), getattr(module, 'weight', None))
+            SCORE_CHANGED.send(
+                sender=None,
+                points_possible=points_possible,
+                points_earned=0,
+                user=student,
+                course_id=course_id,
+                usage_id=module_state_key
+            )
```

Comment:
Nit: can you create an internal helper method for all of the above code, so it's easier to read the top-level function?